New walls, floors and windows

CakeQ opened the pull request:
Replaces walls, floors and windows with my own sprites. 

Also makes WindowComponent a child of IconSmoothComponent so that it can be more flexible, as well as gave windows a non-framed sprite set. Allows for windows with colors now.

~~Modifies Saltern station with a vague flooring layout~~ Will map in flooring in a seperate PR

Pics:
![image](https://user-images.githubusercontent.com/17550641/83158454-b52c3f00-a0fc-11ea-9b2a-f1e8890161c5.png)

![image](https://user-images.githubusercontent.com/17550641/83158513-c6754b80-a0fc-11ea-9cbd-67baa431372d.png)

![image](https://user-images.githubusercontent.com/17550641/83158580-d9881b80-a0fc-11ea-9e6c-ead21488effe.png)

![image](https://user-images.githubusercontent.com/17550641/83158623-e86ece00-a0fc-11ea-9059-3c3a7b2d87a2.png)

![image](https://user-images.githubusercontent.com/17550641/83158649-f1f83600-a0fc-11ea-9a7d-98cb9aea805c.png)

![image](https://user-images.githubusercontent.com/17550641/83158685-fc1a3480-a0fc-11ea-8174-54f8c84906df.png)

SweptWasTaken commented:
Hmm.

For what purpose?

CakeQ commented:
> 
> 
> Hmm.
> 
> For what purpose?

because I wanted to port my turf sprites into ss14

and also because I intend on porting my airlocks as well as further working on the turf set to portray 3/4 view better

As the one who created the turf set I believe there is a stronger coherency across the assets and the work well together, though I'm aware that's just my opinion

CakeQ commented:
Airlock sprites for comparison:
![](https://wiki.baystation12.net/images/1/1b/Headern3.gif)

AJCM-git commented:
I like it, the only criticism i have so far is that the windows dont merge and that the floor is too dark. Maybe keep Eris floor sprites?

Hugal31 commented:
I'm not sure I like profile angle for the airlocks

AJCM-git commented:
Thats the nostalgia talking, perspective airlocks are superior

SweptWasTaken commented:
![Content Client_2020-05-27_11-15-43](https://user-images.githubusercontent.com/49448379/83057727-d6caef00-a046-11ea-9b75-2e02858d5355.png)
![Content Client_2020-05-27_11-16-10](https://user-images.githubusercontent.com/49448379/83057742-db8fa300-a046-11ea-87c5-9ca160b80636.png)
![Content Client_2020-05-27_11-20-40](https://user-images.githubusercontent.com/49448379/83057750-dd596680-a046-11ea-93e9-0598f38a6e49.png)

CakeQ commented:
more progress to come

CakeQ commented:
Updated PR based off of feedback, as well as implemented the rest of the wall layering.

Added two new paint layers to walls for better detail, as well as edge pieces that connect between windows, walls, and airlocks.

Updated previews:
![image](https://user-images.githubusercontent.com/17550641/83158454-b52c3f00-a0fc-11ea-9b2a-f1e8890161c5.png)

![image](https://user-images.githubusercontent.com/17550641/83158513-c6754b80-a0fc-11ea-9cbd-67baa431372d.png)

![image](https://user-images.githubusercontent.com/17550641/83158580-d9881b80-a0fc-11ea-9e6c-ead21488effe.png)

Some further closeups:

![image](https://user-images.githubusercontent.com/17550641/83158623-e86ece00-a0fc-11ea-9059-3c3a7b2d87a2.png)

![image](https://user-images.githubusercontent.com/17550641/83158649-f1f83600-a0fc-11ea-9a7d-98cb9aea805c.png)

![image](https://user-images.githubusercontent.com/17550641/83158685-fc1a3480-a0fc-11ea-8174-54f8c84906df.png)

CakeQ commented:
Updated colors cause PJB wasn't happy
![image](https://user-images.githubusercontent.com/17550641/83170393-5c64a280-a10c-11ea-8698-f74248b3bc04.png)

![image](https://user-images.githubusercontent.com/17550641/83170425-6b4b5500-a10c-11ea-80be-6e91b060f551.png)

ComicIronic commented:
I agree that the lighter wall colours look better.

CakeQ commented:
Forgot to mention that I also added wooden walls: 
![image](https://user-images.githubusercontent.com/17550641/83206435-d0bd3700-a148-11ea-9802-e1252b33da35.png)

CakeQ commented:
PR no longer requires engine changes

ComicIronic commented on Content.Client/GameObjects/Components/IconSmoothing/IconSmoothComponent.cs:
```diff
@@ -41,13 +44,19 @@ public class IconSmoothComponent : Component
         /// <summary>
         ///     We will smooth with other objects with the same key.
         /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
```
Should be readonly.

ComicIronic commented on Content.Client/GameObjects/Components/WindowComponent.cs:
```diff
@@ -1,79 +1,141 @@
+            foreach (var entity in candidates)
             {
-                return;
+                if (entity.TryGetComponent(out IconSmoothComponent other))
```
Should this not be negated?

CakeQ replied:
It should

ComicIronic commented:
All in all, I think it's a real visual improvement.

ComicIronic commented on Content.Client/GameObjects/Components/LowWallComponent.cs:
```diff
@@ -33,6 +35,20 @@ public class LowWallComponent : IconSmoothComponent
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _overlayColor, "overlayColor", null);
```
I *believe* the serializer will do to/from hex for you automatically if you give it a field which has type `Color`, so it should be possible to make your `string` fields `Color`s instead and cut out `FromHex` everywhere you've used it.

CakeQ replied:
Works

PJB3005 commented:
This is completely a subjective change so I'm gonna leave this one up to the emojiocracy on the main PR.

SweptWasTaken commented:
I personally don't like these walls replacing our current walls. 

They look good no doubt but in my opinion our old ones look better

If this pr doesn't get passed we could still use these walls at Nanotransen or somewhere...
